lib/os/mem_blocks.c

```diff
@@ -109,6 +109,52 @@ int sys_mem_blocks_alloc(sys_mem_blocks_t *mem_block, size_t count,
 	return ret;
 }
 
+int sys_mem_blocks_get(sys_mem_blocks_t *mem_block, size_t pos, size_t count, void **out_block)
+{
+	int ret = 0;
+	void *ptr;
+
+	if ((mem_block == NULL) || (out_block == NULL)) {
```

Comment:
Same comment about __ASSERT vs. runtime error for "should never happen" conditions like API misuse by the caller.